### PR TITLE
Oil Burning Locomotive

### PR DESCRIPTION
**Pull request by peternewell**
Add code changes to support oil burning locomotive.

https://blueprints.launchpad.net/or/+spec/steam-locomotive-fuel-types

In draft only at the moment.

ET Discussion Threads -

Wood Fired - 

https://www.elvastower.com/forums/index.php?/topic/38144-steam-locomotive-wood-firing/page__pid__309797#entry309797

Oil Fired -

https://www.elvastower.com/forums/index.php?/topic/38147-steam-locomotive-oil-fired/

### CODE REVIEW

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSLocomotive.cs [hunk 1]
```diff
@@ -239,6 +239,25 @@ public float CurrentLocomotiveSteamHeatBoilerWaterCapacityL
         //float DebugSpeed = 5.0f; // Initialise at 5 mph
         //float DebugTimer = 0.0f;
 
+        public enum SteamLocomotiveFuelTypes
+        {
+            Unknown,
+            Oil,
+            Wood, // not used at the moment
```
Comment (cjakeman) [inline Source/Orts.Simulation/Simulation/RollingStocks/MSTSLocomotive.cs:246]:
```suggestion
            Wood,
```
I'm guessing this comment no longer applies.

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSLocomotive.cs [hunk 2]
```diff
@@ -239,6 +239,25 @@ public float CurrentLocomotiveSteamHeatBoilerWaterCapacityL
         //float DebugSpeed = 5.0f; // Initialise at 5 mph
         //float DebugTimer = 0.0f;
 
+        public enum SteamLocomotiveFuelTypes
```
Comment (cjakeman) [inline Source/Orts.Simulation/Simulation/RollingStocks/MSTSLocomotive.cs:242]: Lines 242-259 only apply to steam so might be better in MSTSSteamLocomotive.cs

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs [hunk 3]
```diff
@@ -1985,13 +1990,21 @@ public override void Update(float elapsedClockSeconds)
                         // amount of water defined in the ENG file, and adding the water defined in the WAG file.
                         float TempMaxCombinedWater = TendersSteamLocomotive.MaxTotalCombinedWaterVolumeUKG;
                         TendersSteamLocomotive.MaxTotalCombinedWaterVolumeUKG = (TempMaxCombinedWater - (Kg.ToLb(TendersSteamLocomotive.MaxLocoTenderWaterMassKG) / WaterLBpUKG)) + (Kg.ToLb(TenderWagonMaxWaterMassKG) / WaterLBpUKG);
-
-                        TendersSteamLocomotive.MaxTenderCoalMassKG = TenderWagonMaxCoalMassKG;
+                                                
                         TendersSteamLocomotive.MaxLocoTenderWaterMassKG = TenderWagonMaxWaterMassKG;
 
+                        if (TendersSteamLocomotive.SteamLocomotiveFuelType == SteamLocomotiveFuelTypes.Oil)
```
Comment (cjakeman) [inline Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs:1996]:
```suggestion
                        if (TendersSteamLocomotive.IsFuelTypeOil)
```
using the property added to the MSTSSteamLocomotive class

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs [hunk 4]
```diff
@@ -148,7 +149,8 @@ public enum WindowState
         public float TrailLocoResistanceFactor; // Factor to reduce base and wind resistance if locomotive is not leading - based upon original Davis drag coefficients
 
         bool TenderWeightInitialize = true;
```
Comment (cjakeman) [inline Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs:151]: I know this is not part of your commit, but perhaps the sense could be reversed from:

bool TenderWeightInitialize = true;

to

bool TenderWeightInitialized = false;

Elsewhere in Open Rails, "Initialised" and "Initialized" are used, so using the opposite is confusing.

**Review by cjakeman — CHANGES_REQUESTED** [review]
Hi Peter,

Thanks for this extension and I've approved the Blueprint.

I've suggested some changes and also emailed a "diff" to you after exploring them on my PC.

Hope that makes sense.

Chris

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs [hunk 5]
```diff
@@ -52,6 +52,7 @@
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
+using static Orts.Simulation.RollingStocks.MSTSLocomotive;
```
Comment (cjakeman) [inline Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs:55]: Not comfortable with this reference from a parent class back to a child class.
Instead you could delete it and add a property in the child class as I suggested elsewhere.

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSLocomotive.cs (hunk 1 above)
Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSLocomotive.cs]: Ok, thanks for identifying this.

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSLocomotive.cs (hunk 2 above)
Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSLocomotive.cs]: Ok, will do.

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs (hunk 4 above)
Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs]: Ok, done.

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs (hunk 5 above)
Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs]: Hopefully this has been corrected with changes that I have added.

**Comment by peternewell** [comment]
> 
> Thanks for this extension and I've approved the Blueprint.
> 
> I've suggested some changes and also emailed a "diff" to you after exploring them on my PC.
> 
> Hope that makes sense.
> 

Thanks for that. I have made a couple of hopefully small changes to your suggestions, for consistency, which will be acceptable.

**Review by cjakeman — APPROVED** [review]
Hi Peter,

Thanks for adopting (and improving on) my changes.

I'll leave you to do the merge.

Best wishes,

Chris

**Comment by peternewell** [comment]
Chris,

Thanks for that.

**Pull request merged**